openssl-dynamic/src/main/c/ssl_private.h

```diff
@@ -353,6 +353,9 @@ struct tcn_ssl_task_t {
     jobject task;
 };
 
+tcn_ssl_task_t* tcn_ssl_task_new(JNIEnv*, jobject);
```

Comment:
didn't use tcn_ssl_task_new in the end?

Reply:
its only used by `SSLContext.c` atm in which we also define it. That said I thought it is cleaner to have the new and free here :)